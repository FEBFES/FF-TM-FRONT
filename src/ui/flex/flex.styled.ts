import styled from 'styled-components';
import { DirectionType, JustyfyContentType, AlignItemsType } from './flex';

interface SFlexContainerProps {
  dir: DirectionType;
  jc: JustyfyContentType;
  ai: AlignItemsType;
}

export const SFlexContainer = styled.div<SFlexContainerProps>`
  display: flex;
  flex-direction: ${({ dir }) => dir};
  align-items: ${({ ai }) => ai};
  justify-content: ${({ jc }) => jc};
`;
