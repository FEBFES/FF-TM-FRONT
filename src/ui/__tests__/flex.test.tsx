import React from 'react';
import { Flex } from '../flex/flex';
import renderer from 'react-test-renderer';

describe('<Button />', () => {
  it('SNAPSHOTS: UI - Flex (default) component - render', () => {
    const elem = renderer
      .create(
        <Flex>
          <div>test</div>
        </Flex>
      )
      .toJSON();
    expect(elem).toMatchSnapshot();
  });

  it('SNAPSHOTS: UI - Flex (center) component - render', () => {
    const elem = renderer
      .create(
        <Flex ai={'center'} jc={'center'}>
          <div>test</div>
        </Flex>
      )
      .toJSON();
    expect(elem).toMatchSnapshot();
  });

  it('SNAPSHOTS: UI - Flex (start) component - render', () => {
    const elem = renderer
      .create(
        <Flex ai={'start'} jc={'start'}>
          <div>test</div>
        </Flex>
      )
      .toJSON();
    expect(elem).toMatchSnapshot();
  });

  it('SNAPSHOTS: UI - Flex (end) component - render', () => {
    const elem = renderer
      .create(
        <Flex ai={'end'} jc={'end'}>
          <div>test</div>
        </Flex>
      )
      .toJSON();
    expect(elem).toMatchSnapshot();
  });

  it('SNAPSHOTS: UI - Flex (between) component - render', () => {
    const elem = renderer
      .create(
        <Flex ai={'between'} jc={'between'}>
          <div>test</div>
        </Flex>
      )
      .toJSON();
    expect(elem).toMatchSnapshot();
  });

  it('SNAPSHOTS: UI - Flex (around) component - render', () => {
    const elem = renderer
      .create(
        <Flex ai={'around'} jc={'around'}>
          <div>test</div>
        </Flex>
      )
      .toJSON();
    expect(elem).toMatchSnapshot();
  });
});
