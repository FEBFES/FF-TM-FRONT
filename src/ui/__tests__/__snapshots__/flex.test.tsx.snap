// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Flex /> SNAPSHOTS: UI - Flex (around) component - render 1`] = `
<div
  ai="around"
  className="sc-bcPKhP ffArlj"
  dir="row"
  jc="around"
>
  <div>
    test
  </div>
</div>
`;

exports[`<Flex /> SNAPSHOTS: UI - Flex (between) component - render 1`] = `
<div
  ai="between"
  className="sc-bcPKhP bwWTUT"
  dir="row"
  jc="between"
>
  <div>
    test
  </div>
</div>
`;

exports[`<Flex /> SNAPSHOTS: UI - Flex (center) component - render 1`] = `
<div
  ai="center"
  className="sc-bcPKhP ksstrX"
  dir="row"
  jc="center"
>
  <div>
    test
  </div>
</div>
`;

exports[`<Flex /> SNAPSHOTS: UI - Flex (default) component - render 1`] = `
<div
  ai="center"
  className="sc-bcPKhP cGhUzG"
  dir="row"
  jc="between"
>
  <div>
    test
  </div>
</div>
`;

exports[`<Flex /> SNAPSHOTS: UI - Flex (end) component - render 1`] = `
<div
  ai="end"
  className="sc-bcPKhP ilOKvr"
  dir="row"
  jc="end"
>
  <div>
    test
  </div>
</div>
`;

exports[`<Flex /> SNAPSHOTS: UI - Flex (start) component - render 1`] = `
<div
  ai="start"
  className="sc-bcPKhP jmJEXb"
  dir="row"
  jc="start"
>
  <div>
    test
  </div>
</div>
`;
